src/makeCalculator.js

```diff
@@ -4,7 +4,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(num) {
+      this.result += num;
+    },
+
+    subtract(num) {
+      this.result -= num;
+    },
+
+    multiply(num) {
+      this.result *= num;
+    },
+
+    divide(num) {
+      if (num !== 0) {
+        this.result /= num;
+      }
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, num) {
```

Comment:
The `operate` method should check if the provided callback is a function before attempting to call it. This can be done using `typeof callback === 'function'` to prevent runtime errors if a non-function is passed.